File: 125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = "".join([ch for ch in s if ch.isalpha() or ch.isdigit()]).lower()
        return s == s[::-1]

class Solution:
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        import string
        if not s:
            return True
        else:
            punc = set(string.punctuation)
            s = ''.join(ch for ch in s if ch not in punc)
            s = s.replace(' ','')
            s = s.lower()
            if s == s[::-1]:
                return True
            else:
                return False

File: 125. Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s==null || s.length()==0) {
            return true;
        }
        char[] cs = s.toLowerCase().toCharArray();
        int write = 0;
        for (int read=0; read<s.length(); read++) {
            if ((cs[read] - 'a' < 26 && cs[read] - 'a' >= 0)||(cs[read] - '0' < 10 && cs[read] - '0' >= 0)) {
                cs[write++] = cs[read];
            }
        }
        int st = 0, ed = write-1;
        while (st < ed) {
            if (cs[st++] != cs[ed--]) {
                return false;
            }
        }
        return true;
    }
}